components/AnimatedLoading.js

```diff
@@ -0,0 +1,85 @@
+import React, { useCallback, useEffect, useRef } from 'react';
+import { Animated } from 'react-native';
+import LottieView from 'lottie-react-native';
+import PropTypes from 'prop-types';
+import { connectStyle } from '@shoutem/theme';
+import { View } from '@shoutem/ui';
+import { animations } from '../assets';
+import { usePreviousValue } from '../hooks';
+
+const AnimatedView = Animated.createAnimatedComponent(View);
+
+function AnimatedLoading({
+  children,
+  finishAnimationCallback,
+  loading,
+  style,
+}) {
+  const prevLoading = usePreviousValue(loading);
+  const animateProgress = useRef(new Animated.Value(0)).current;
+
+  useEffect(() => {
+    if (!prevLoading && loading) {
+      startAnimation();
+    }
+  }, [prevLoading, startAnimation, loading]);
+
+  const startAnimation = useCallback(() => {
+    Animated.timing(animateProgress, {
+      toValue: 1,
+      duration: 300,
+      useNativeDriver: false,
+    }).start(finishAnimation);
+  }, [animateProgress, finishAnimation]);
+
+  const finishAnimation = useCallback(() => {
```

Comment:
am i crazy, or these animated values aren't used for anything right now ? You are simply conditionally rendering the lottie animation, which doesn't require you to animate anything in particular